app/models/concerns/episode_media.rb

```diff
@@ -111,39 +192,119 @@ def video_content_type?(*args)
   end
 
   def media_duration
-    media.inject(0.0) { |s, c| s + c.duration.to_f } + podcast.try(:duration_padding).to_f
+    if override?
+      external_media_resource&.duration
+    else
+      media.inject(0.0) { |s, c| s + c.duration.to_f } + podcast.try(:duration_padding).to_f
+    end
   end
 
   def media_file_size
-    media.inject(0) { |s, c| s + c.file_size.to_i }
+    if override?
+      external_media_resource&.file_size
+    else
+      media.inject(0) { |s, c| s + c.file_size.to_i }
+    end
   end
 
+  # 1) must_be_complete or not
+  # that is called with "true" in 2 places
+  # - in checking valid on 1st time publish
+  # - displaying "complete" for status
+  # So really this is about 1st time, must be fully ready,
+  # as there is no prior published media version to fall back on?
+  # And for status display or api responses, are media in the current version done?
+  #
+  # 2) override or not
+  # If there is an override, that is the only status to check
+  # We need a (different?) method to check contents ready for media versions
+  # but if an override is provided, we don't look at uploaded media status (uncut or contents)
+  #
+  # 3) enough segments
+  # Do we have enough media files for the segment count?
+  # That should get checked regardless of must_be_complete
+  #
+  # 4) uncut present
+  # If this is not must_be_complete, then
+  # If a single file upload, is there file uploaded, that isn't marked deleted
+  #
+  # 5) TODO: invalid or error status?
+  # should counts of files include invalid/error status files?
+  # should it be ready if any files is in error or invalid state?
   def media_ready?(must_be_complete = true)
-    if media.empty?
+    if !must_be_complete && medium_uncut?
+      uncut.present? && !uncut.marked_for_destruction?
```

Comment:
Seems like this was another bug, right?  Maybe related to #1164?

Reply:
idk that I was thinking about that bug, but I was incorporating the uncut logic that was in the validate method into this media_ready? method, but I think it might fix that issue - I need to test it more to be sure.